ui/app/components/secret-engine/configure-azure.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import errorMessage from 'vault/utils/error-message';
+
+import type ConfigModel from 'vault/models/azure/config';
+import type IdentityOidcConfigModel from 'vault/models/identity/oidc/config';
+import type Router from '@ember/routing/router';
+import type StoreService from 'vault/services/store';
+import type VersionService from 'vault/services/version';
+import type FlashMessageService from 'vault/services/flash-messages';
+
+/**
+ * @module SecretEngineConfigureAzure component is used to configure the Azure secret engine
+ * For enterprise users, they will see an additional option to config WIF attributes in place of Azure account attributes.
+ * If the user is configuring WIF attributes they will also have the option to update the global issuer config, which is a separate endpoint named identity/oidc/config.
+ * @example
+ * <SecretEngine::ConfigureAzure
+    @model={{this.model.azure-config}}
+    @backendPath={{this.model.id}}
+    @issuerConfig={{this.model.identity-oidc-config}}
+    />
+ * 
+ * @param {object} model - Azure config model
+ * @param {string} backendPath - name of the Azure secret engine, ex: 'azure-123'
+ * @param {object} issuerConfigModel - the identity/oidc/config model
+ */
+
+interface Args {
+  model: ConfigModel;
+  issuerConfig: IdentityOidcConfigModel;
+  backendPath: string;
+}
+
+export default class ConfigureAzureComponent extends Component<Args> {
+  @service declare readonly router: Router;
+  @service declare readonly store: StoreService;
+  @service declare readonly version: VersionService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked accessType = 'azure';
+  @tracked errorMessage = '';
+  @tracked invalidFormAlert = '';
+  @tracked saveIssuerWarning = '';
+
+  disableAccessType = false;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    // the following checks are only relevant to existing enterprise configurations
+    if (this.version.isCommunity && this.args.model.isNew) return;
```

Comment:
Claire, I know you had left a comment to reverse this, but my brain —not remembering you had asked to flip it, reversed it back—to me it reads easier without negative logic. So "you're community and you're a new model" this is not relevant to you, instead of "you're _not_ enterprise and you're _not_ a new model proceed." 

I think you're point is that you just want to know you're editing then proceed, but I tend to read the logic statement verbatim instead of recognizing a state the user is in, like editing. 

It's the same logic statement for AWS at the moment as well (an easy change if need be).

Reply:
I definitely agree with making logic readable. To me, your comment above is pretty much: 
```js
if (this.version.isEnterprise && !this.args.model.isNew)
```
So that's why I thought swapping the isNew felt clearer. But potato potato, doesn't matter to me. 🥔 😄